### PR TITLE
sqlserver: limit concurrent database operations

### PR DESCRIPTION
**Pull request by mialinx**
### SQLServer

Limit number of concurrent database operations (eg backups)

Try to return http error to SQLServer on proxy panic

### CODE REVIEW

File: internal/config.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ const (
 	SQLServerBlobKeyFile      = "SQLSERVER_BLOB_KEY_FILE"
 	SQLServerBlobLockFile     = "SQLSERVER_BLOB_LOCK_FILE"
 	SQLServerConnectionString = "SQLSERVER_CONNECTION_STRING"
+	SQLServerDBConcurency     = "SQLSERVER_DB_CONCURENCY"
```
Comment (usernamedt) [inline internal/config.go:112]: s/CONCURENCY/CONCURRENCY/g
s/SQLServerDBConcurency/SQLServerDBConcurrency/g

File: internal/databases/sqlserver/sqlserver.go [hunk 2]
```diff
@@ -398,6 +404,15 @@ func runParallel(f func(int) error, cnt int) error {
 	return nil
 }
 
+func getDBConcurrency() int {
+	concurrency, err := internal.GetMaxConcurrency(internal.SQLServerDBConcurency)
+	if err != nil {
+		tracelog.WarningLogger.Printf("config error: %v", err)
```
Comment (usernamedt) [inline internal/databases/sqlserver/sqlserver.go:410]: maybe also log the chosen `blob.DefaultConcurency` value?

File: internal/databases/sqlserver/sqlserver.go [hunk 3]
```diff
@@ -378,10 +378,16 @@ func getLogsSinceBackup(folder storage.Folder, backupName string, stopAt time.Ti
 	return logNames, nil
 }
 
-func runParallel(f func(int) error, cnt int) error {
+func runParallel(f func(int) error, cnt int, concurency int) error {
+	if concurency == 0 {
+		concurency = cnt
+	}
+	sem := make(chan struct{}, concurency)
```
Comment (usernamedt) [inline internal/databases/sqlserver/sqlserver.go:385]: semaphore.NewWeighted?

File: internal/databases/sqlserver/sqlserver.go [hunk 4]
```diff
@@ -378,10 +378,16 @@ func getLogsSinceBackup(folder storage.Folder, backupName string, stopAt time.Ti
 	return logNames, nil
 }
 
-func runParallel(f func(int) error, cnt int) error {
+func runParallel(f func(int) error, cnt int, concurency int) error {
+	if concurency == 0 {
```
Comment (usernamedt) [inline internal/databases/sqlserver/sqlserver.go:382]: just in case :)
concurrency <= 0 ?

File: internal/databases/sqlserver/sqlserver.go (hunk 3 above)
Comment (mialinx) [reply internal/databases/sqlserver/sqlserver.go:385]: chans are idiomatic for simple semaphores, and we don't need weights here

**Review by usernamedt — APPROVED** [review]

**Pull request merged**